homeassistant/core.py

```diff
@@ -1260,6 +1263,8 @@ def __init__(
         self.domain, self.object_id = split_entity_id(self.entity_id)
         self._as_dict: ReadOnlyDict[str, Collection[Any]] | None = None
         self._as_compressed_state: dict[str, Any] | None = None
```

Comment:
I think we can drop the unserialized cache in a future PR to save a small bit of RAM but need to verify it's always covered by the serialized cache.